dsgrid/utils/dataset.py

```diff
@@ -402,10 +402,23 @@ def repartition_if_needed_by_mapping(
     df: DataFrame,
     mapping_type: DimensionMappingType,
     scratch_dir_context: ScratchDirContext,
-) -> DataFrame:
-    """Repartition the dataframe if the mapping might cause data skew."""
+    repartition: bool | None = None,
+) -> tuple[DataFrame, Path | None]:
+    """Repartition the dataframe if the mapping might cause data skew.
```

Comment:
Define data skew? (I don't recall exactly what you mean by the term ... something about natural partitions being vastly different sizes???)